docs/repository-management.md

```diff
@@ -115,6 +115,8 @@ A pull request can generally be merged once it is:
 
 The final pull request merge decision is made by the **@wordpress/gutenberg-core** team.
 
+Please make sure to assign your merged pull request to its release milestone. Doing so creates the historical legacy of what code landed when, and makes it possible for all project contributors (even non-technical ones) to access this information.
```

Comment:
To be clear, this doesn't "create" a history; "surfaces" might be more accurate phrasing. The history already exists regardless, and the milestoning could be retroactively applied at any time (which may be interesting for the thousands of existing pull requests which could be milestoned to existing releases).